src/network/networkd-ndisc.c

```diff
@@ -1035,18 +1047,19 @@ int config_parse_address_generation_type(
 
         token->prefix = buffer.in6;
 
-        r = ordered_hashmap_ensure_allocated(&network->ipv6_tokens, &ipv6_token_hash_ops);
+        r = ordered_set_ensure_allocated(&network->ipv6_tokens, &ipv6_token_hash_ops);
         if (r < 0)
                 return log_oom();
 
-        r = ordered_hashmap_put(network->ipv6_tokens, &token->prefix, token);
-        if (r < 0) {
+        r = ordered_set_put(network->ipv6_tokens, token);
+        if (r == -EEXIST)
```

Comment:
Same here.

Reply:
But here, the code is correct, as this is `ordered_set_put()`.

Reply:
Hmmm, see `test-ordered-set.c`, `test_set_put()`. The return value is 0.

Reply:
Please see #16561.

Reply:
Right, so we get -EEXIST because `ipv6_token_hash_ops != `trivial_hash_ops`. I think we should fix this at some point, but not in this PR.